tuftool/src/rhtas.rs

```diff
@@ -0,0 +1,343 @@
+// Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: MIT OR Apache-2.0
+
+use crate::build_targets;
+use crate::common::UNUSED_URL;
+use crate::datetime::parse_datetime;
+use crate::error::{self, Result};
+use crate::source::parse_key_source;
+use chrono::{DateTime, Utc};
+use clap::Parser;
+use serde_json::json;
+use snafu::{OptionExt, ResultExt};
+use std::num::NonZeroU64;
+use std::path::{Path, PathBuf};
+use tough::editor::RepositoryEditor;
+use tough::{ExpirationEnforcement, RepositoryLoader};
+use url::Url;
+
+#[derive(Debug, Parser)]
+pub(crate) struct RhtasArgs {
+    /// Allow repo download for expired metadata
+    #[arg(long)]
+    allow_expired_repo: bool,
+
+    /// Follow symbolic links in the given directory when adding targets
+    #[arg(short, long)]
+    follow: bool,
+
+    /// Incoming metadata from delegatee
+    #[arg(short, long = "incoming-metadata")]
+    indir: Option<Url>,
+
+    /// Key files to sign with
+    #[arg(short, long = "key", required = true)]
+    keys: Vec<String>,
+
+    /// TUF repository metadata base URL
+    #[arg(short, long = "metadata-url")]
+    metadata_base_url: Url,
+
+    /// The directory where the updated repository will be written
+    #[arg(short, long)]
+    outdir: PathBuf,
+
+    /// Path to root.json file for the repository
+    #[arg(short, long)]
+    root: PathBuf,
+
+    /// Role of incoming metadata
+    #[arg(long)]
+    role: Option<String>,
+
+    /// Expiration of snapshot.json file; can be in full RFC 3339 format, or something like 'in
+    /// 7 days'
+    #[arg(long, value_parser = parse_datetime)]
+    snapshot_expires: DateTime<Utc>,
+
+    /// Version of snapshot.json file
+    #[arg(long)]
+    snapshot_version: NonZeroU64,
+
+    /// Path to the new Fulcio target file to add to the targets
+    #[arg(long = "set-fulcio-target")]
+    fulcio_target: Option<PathBuf>,
+
+    /// Status for the Fulcio target
```

Comment:
All the status flags should only allow setting the status to "Active" or "Expired". Right now, I can provide arbitrary string (e.g. "foobar") and the command allows me to do that, which it shouldn't.